mysql-test/suite/component_masking_functions/t/flusher_thread_connection_reuse.test

```diff
@@ -0,0 +1,40 @@
+--source include/have_debug.inc
+--source include/have_debug_sync.inc
+--source include/have_masking_functions_component.inc
+
+--source include/count_sessions.inc
+
+CREATE TABLE mysql.masking_dictionaries(
+    Dictionary VARCHAR(256) NOT NULL,
+    Term VARCHAR(256) NOT NULL,
+    UNIQUE INDEX dictionary_term_idx (Dictionary, Term)
+) ENGINE = InnoDB DEFAULT CHARSET=utf8mb4;
+GRANT SELECT, INSERT, UPDATE, DELETE ON mysql.masking_dictionaries TO 'mysql.session'@'localhost';
+INSERT INTO mysql.masking_dictionaries VALUES('dict1', 'term1');
+
+
+SET GLOBAL debug = '+d,enable_masking_functions_flusher_create_sync,enable_masking_functions_flush_thread_sync,enable_masking_functions_flush_thread_double_pass';
+SET debug_sync = 'masking_functions_after_flusher_create WAIT_FOR masking_functions_after_flusher_create_signal';
+
+send INSTALL COMPONENT 'file://component_masking_functions';
+
+--connect(con_priv,localhost,root,,)
+SET debug_sync = 'now SIGNAL masking_functions_before_cache_reload_signal WAIT_FOR masking_functions_after_cache_reload_signal';
+
+SET debug_sync = 'now SIGNAL masking_functions_after_flusher_create_signal';
+
+--connection default
+--reap
+
+--disconnect con_priv
+
+SET debug_sync = 'now SIGNAL masking_functions_before_cache_reload_signal WAIT_FOR masking_functions_after_cache_reload_signal';
+
+--source wait_for_component_uninstall.inc
+
+SET GLOBAL debug = '-d,enable_masking_functions_flusher_create_sync,enable_masking_functions_flush_thread_sync,enable_masking_functions_flush_thread_double_pass';
```

Comment:
I think it is a good idea to reset debug_sync facility at the end of the test as well.

Reply:
Done